sql/src/main/java/io/crate/analyze/QueriedTable.java

```diff
@@ -92,4 +96,155 @@ public WhereClause consumeWhereClause(WhereClause clause) {
     public OrderBy consumeOrderBy(OrderBy orderBy) {
         return null;
     }
+
+    public static QueriedTable newSubRelation(QualifiedName name, TableRelation tableRelation, QuerySpec querySpec) {
+        QuerySpec splitQuerySpec = new QuerySpec();
+        List<Symbol> outputs = querySpec.outputs();
+        if (outputs == null || outputs.isEmpty()) {
+            throw new IllegalArgumentException("a querySpec needs to have some outputs in order to create a new sub-relation");
+        }
+        splitQuerySpec.outputs(splitForRelation(tableRelation, outputs).splitSymbols());
+
+
+        WhereClause where = querySpec.where();
+        if (where != null && where.hasQuery()) {
+            QuerySplitter.SplitQueries splitQueries = QuerySplitter.splitForRelation(where.query(), tableRelation);
```

Comment:
maybe keep the interface the same as above: splitForRelation(relation, thingToSplit)